From f4fbb07b993b6c782c470e88b6c09f8484bd61fa Mon Sep 17 00:00:00 2001
From: Aswin M <Aswin_M@comcast.com>
Date: Wed, 15 Jun 2022 12:34:15 +0000
Subject: [PATCH] HX44X-149 : Bring in Voice Support

Reason for change: Add support for external JS functions from the GUI Client.
Test Procedure: Enable necessary configurations and build with avs-sdk-oobe-screens.
Risks: Low
License: Inherited
Upstream-Status: Pending
Source: RDKM

Signed-off-by: Aswin M <Aswin_M@comcast.com>
---
 modules/GUI/js/src/app.tsx   | 6 ++++++
 modules/GUI/js/tsconfig.json | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/modules/GUI/js/src/app.tsx b/modules/GUI/js/src/app.tsx
index 6360608..87ccdd5 100644
--- a/modules/GUI/js/src/app.tsx
+++ b/modules/GUI/js/src/app.tsx
@@ -238,6 +238,9 @@ export class App extends React.Component<any, IAppState> {
     }
 
     protected handleRequestAuthorization(requestAuthorizationMessage : IRequestAuthorizationMessage) {
+
+        window['onHandleRequestAuthorization'](requestAuthorizationMessage);
+
         /**
          * Use to present CBL authorization.
          * API :
@@ -248,6 +251,9 @@ export class App extends React.Component<any, IAppState> {
     }
 
     protected handleAuthorizationStateChanged(authStateChangeMessage : IAuthorizationChangeMessage) {
+
+        window['onHandleAuthorizationStateChanged'](authStateChangeMessage);
+
         // Use to drive app behavior based on authorization state changes.
     }
 
diff --git a/modules/GUI/js/tsconfig.json b/modules/GUI/js/tsconfig.json
index ec512c8..a54cd93 100644
--- a/modules/GUI/js/tsconfig.json
+++ b/modules/GUI/js/tsconfig.json
@@ -21,7 +21,8 @@
     "jsx": "react",
     "types": [
       "node"
-    ]
+    ],
+    "suppressImplicitAnyIndexErrors": true
   },
   "exclude": [
     "node_modules",
-- 
2.17.1

